src/System.Net.WebSockets.Client/tests/ClientWebSocketTest.cs

```diff
@@ -27,18 +29,637 @@ public ClientWebSocketTest(ITestOutputHelper output)
+                    for (int i = 0; i < tasks.Length; i++)
+                    {
+                        tasks[i] = cws.SendAsync(
+                            WebSocketData.GetBufferFromText("hello"), 
+                            WebSocketMessageType.Text, 
+                            true, 
+                            cts.Token);
+                    }
+
+                    Task.WaitAll(tasks);
+
+                    Assert.Equal(WebSocketState.Open, cws.State);
+                }
+                catch (AggregateException ag)
+                {
+                    foreach (var ex in ag.InnerExceptions)
+                    {
+                        if (ex is InvalidOperationException)
+                        {
+                            Assert.Equal(
+                                ResourceHelper.GetExceptionMessage(
+                                    "net_Websockets_AlreadyOneOutstandingOperation", 
+                                    "SendAsync"),
+                                ex.Message);
+
+                            Assert.Equal(WebSocketState.Aborted, cws.State);
+                        }
+                        else if (ex is WebSocketException)
+                        {
+                            // Multiple cases.
+                            Assert.Equal(WebSocketState.Aborted, cws.State);
+
+                            WebSocketError errCode = (ex as WebSocketException).WebSocketErrorCode;
+                            Assert.True(
+                                (errCode == WebSocketError.InvalidState) || (errCode == WebSocketError.Success),
+                                "WebSocketErrorCode");
+                        }
+                        else
+                        {
+                            Assert.True(false, "Unexpected exception: " + ex.Message);
+                        }
+                    }
+                }
+            }
+        }
+
+        [ConditionalTheory("WebSocketsSupported"), MemberData("EchoServers")]
+        public async Task ReceiveAsync_MultipleOutstandingReceiveOperations_Throws(Uri server)
+        {
+            using (ClientWebSocket cws = await WebSocketHelper.GetConnectedWebSocket(server, TimeOutMilliseconds, _output))
+            {
+                var cts = new CancellationTokenSource(TimeOutMilliseconds);
+
+                Task[] tasks = new Task[2];
+
+                await cws.SendAsync(
+                    WebSocketData.GetBufferFromText(".delay5sec"), 
+                    WebSocketMessageType.Text, 
+                    true, 
+                    cts.Token);
+
+                var recvBuffer = new byte[100];
+                var recvSegment = new ArraySegment<byte>(recvBuffer);
+
+                try
+                {
+                    for (int i = 0; i < tasks.Length; i++)
+                    {
+                        tasks[i] = cws.ReceiveAsync(recvSegment, cts.Token);
+                    }
+
+                    Task.WaitAll(tasks);
```

Comment:
Ah, I just noticed you might be using WaitAll because you want all of the exceptions, not just one of them?